### PR TITLE
ENH add fetcher for John Hopkins dataset

### LINKED ISSUE #11: Grab data directly from John Hopkins
We are using pycovid so far. However, it is itself pulling from an R package which mirrors a tidied version of the John Hopkins data. As a consequence, we are lagging behind.

The biggest added value of pycovid is the merge with the ISO country code (which we need for plotting on the map). I believe that this merge is currently broken for China:
https://github.com/sudharshan-ashok/pycovid/issues/2

I suggest that we pull directly from John Hopkins: they are hosting a version of the data which is formatted close to what we need:
https://github.com/CSSEGISandData/COVID-19/tree/master/csse_covid_19_data/csse_covid_19_time_series

This would entail changing the logic in https://github.com/covid19-dash/covid-dashboard/blob/master/data_input.py

The hardest work would be to extract the ISO country codes from 
https://raw.githubusercontent.com/lukes/ISO-3166-Countries-with-Regional-Codes/master/slim-3/slim-3.csv

### PR DESCRIPTION
**Pull request by glemaitre**
closes #11 

Making a fetcher to replace `pycovid`. The idea would be to keep the same column name to change the other in a minimal manner.

### CODE REVIEW

**Comment by glemaitre** [comment]
ping @GaelVaroquaux @emmanuelle I think this should be ready at least for testing.

**Comment by GaelVaroquaux** [comment]
OK, let check out the branch and test. Thanks!!

File: fetcher.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    "Cote d'Ivoire": "Côte d'Ivoire",
+    'Czechia': "Czech Republic",
+    'Eswatini': "Swaziland",
+    'Holy See': "Italy",
+    'Iran': "Iran, Islamic Republic of",
+    'Korea, South': "South Korea",
+    'Moldova': "Moldova, Republic of",
+    'North Macedonia': "Macedonia, the former Yugoslav Republic of",
+    'Reunion': "Réunion",
+    'Taiwan*': "Taiwan",
```
Comment (GaelVaroquaux) [inline fetcher.py:31]: Is the star a regexp? It seems like a good idea to use matching. I just don't recognize the syntax.

File: fetcher.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    )
+
+    assert df["Country"].unique().shape == (141,), \
+        "Missing countries when making the merge"
+
+    df = df.rename(
+        columns={
+            "Country/Region": "name",
+            "Country": "country_region",
+            "Alpha-3 code": "alpha-3",
```
Comment (GaelVaroquaux) [inline fetcher.py]: We're renaming "alpha-3" to "iso" in the input_data function. Maybe we could change this right here.

File: fetcher.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+def fetch_john_hopkins_data():
+    df_covid = {
+        key: pd.read_csv(os.path.join(URL_BASE, filename))
+        for key, filename in FILENAME_JOHN_HOPKINS.items()
+    }
+    columns_drop = ["Province/State", "Lat", "Long"]
+    for key in df_covid:
+        df_covid[key] = df_covid[key].drop(columns=columns_drop)
+        df_covid[key] = df_covid[key].groupby("Country/Region").sum()
+        df_covid[key] = df_covid[key].stack().to_frame().reset_index().rename(
```
Comment (GaelVaroquaux) [inline fetcher.py:58]: I could use a little comment to help me understand what this line does :)

**Comment by GaelVaroquaux** [comment]
I have the impression that we are not retrieving South Korea and China (or I am failing to query them). The absence of these countries is a motivation for not depending on pycovid.

Can you check what is going with them?

File: fetcher.py (hunk 1 above)
Comment (glemaitre) [reply fetcher.py:31]: I assume this because John Hopkins mutualize both Tawain area

**Comment by glemaitre** [comment]
> Can you check what is going with them?

I can get china:


```python
In [10]: df[df["name"] == "China"]                                                                                                                                                                 
Out[10]: 
       name       date  cases       type country_region Alpha-2 code alpha-3  Numeric code   lat   long
4293  China 2020-01-22    548  confirmed          China           CN     CHN         156.0  35.0  105.0
4294  China 2020-01-23    643  confirmed          China           CN     CHN         156.0  35.0  105.0
4295  China 2020-01-24    920  confirmed          China           CN     CHN         156.0  35.0  105.0
4296  China 2020-01-25   1406  confirmed          China           CN     CHN         156.0  35.0  105.0
4297  China 2020-01-26   2075  confirmed          China           CN     CHN         156.0  35.0  105.0
...     ...        ...    ...        ...            ...          ...     ...           ...   ...    ...
4447  China 2020-03-10  60181  recovered          China           CN     CHN         156.0  35.0  105.0
4448  China 2020-03-11  61644  recovered          China           CN     CHN         156.0  35.0  105.0
4449  China 2020-03-12  62901  recovered          China           CN     CHN         156.0  35.0  105.0
4450  China 2020-03-13  64196  recovered          China           CN     CHN         156.0  35.0  105.0
4451  China 2020-03-14  65660  recovered          China           CN     CHN         156.0  35.0  105.0
```

I think this should happen in the `get_data`?

**Comment by GaelVaroquaux** [comment]
OK. China is not displayed in the map, though? I don't see it.

**Comment by GaelVaroquaux** [comment]
Also, a simpler test:
<pre>
confirmed = data_input.get_data()['confirmed']
confirmed['FRA']
confirmed['CHN']
</pre>
The query to France works, but not to China.

These above could almost be tests for the code (as in, a beginning of unit-testing).

**Comment by glemaitre** [comment]
Uhm it works for me

```python
In [1]: from data_input import get_data                                                                                                                                                            

In [2]: df = get_data()                                                                                                                                                                            

In [3]: confirmed = df["confirmed"]                                                                                                                                                                

In [4]: confirmed["CHN"]                                                                                                                                                                           
Out[4]: 
country_region    China
date                   
2020-01-22          548
2020-01-23         1191
2020-01-24         2111
2020-01-25         3517
2020-01-26         5592
2020-01-27         8469
2020-01-28        13978
2020-01-29        20065
2020-01-30        28206
2020-01-31        38008
2020-02-01        49899
2020-02-02        66529
2020-02-03        86245
2020-02-04       109952
2020-02-05       137392
2020-02-06       167979
2020-02-07       202089
2020-02-08       238903
2020-02-09       278732
2020-02-10       321086
2020-02-11       365472
2020-02-12       410231
2020-02-13       470126
2020-02-14       536484
2020-02-15       604897
2020-02-16       675410
2020-02-17       747844
2020-02-18       822055
2020-02-19       896674
2020-02-20       971751
2020-02-21      1047301
2020-02-22      1124302
2020-02-23      1201324
2020-02-24      1278565
2020-02-25      1356319
2020-02-26      1434485
2020-02-27      1513085
2020-02-28      1592013
2020-02-29      1671369
2020-03-01      1751301
2020-03-02      1831437
2020-03-03      1911698
2020-03-04      1992084
2020-03-05      2072621
2020-03-06      2153311
2020-03-07      2234081
2020-03-08      2314904
2020-03-09      2395764
2020-03-10      2476651
2020-03-11      2557572
2020-03-12      2638504
2020-03-13      2719449
2020-03-14      2800426
```

**Comment by GaelVaroquaux** [comment]
OK, I probably had forgotten to restart my kernel when moving to your branch. It works indeed. Sorry!!

**Comment by GaelVaroquaux** [comment]
I can confirm that the input is now fixed. Merging as this is a net improvement. Thank you!!

**Pull request merged**